cmd/podman/unshare.go

```diff
@@ -0,0 +1,63 @@
+// +build linux
+
+package main
+
+import (
+	"os"
+	"os/exec"
+
+	"github.com/containers/buildah/pkg/unshare"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	unshareDescription = "\n  Runs a command in a modified user namespace."
+	unshareCommand     = &cobra.Command{
+		Use:   "unshare",
+		Short: "Run a command in a modified user namespace",
+		Long:  unshareDescription,
+		RunE:  unshareCmd,
+		Example: `podman unshare id
+  podman unshare cat /proc/self/uid_map,
+  podman unshare podman-script.sh`,
+	}
+)
+
+func init() {
+	unshareCommand.SetUsageTemplate(UsageTemplate())
+	flags := unshareCommand.Flags()
+	flags.SetInterspersed(false)
+}
+
+// unshareCmd execs whatever using the ID mappings that we want to use for ourselves
+func unshareCmd(c *cobra.Command, args []string) error {
+	// Set the default isolation type to use the "rootless" method.
+	if _, present := os.LookupEnv("PODMAN_ISOLATION"); !present {
+		if err := os.Setenv("PODMAN_ISOLATION", "rootless"); err != nil {
+			logrus.Errorf("error setting PODMAN_ISOLATION=rootless in environment: %v", err)
+			os.Exit(1)
+		}
+	}
+
+	// force reexec using the configured ID mappings
+	unshare.MaybeReexecUsingUserNamespace(true)
```

Comment:
if you drop this line, would podman setup the namespace automatically?

Reply:
Yes, It sets up the namespace automatically, If I remove this line.